### DISCUSSION TITLE
Which dashboard resource should I use?

### DISCUSSION THREAD

**Discussion by knnthmacpherson** [newrelic/terraform-provider-newrelic#2216] [Q&A] [answered]
Looks like there's 3 options but I don't get which one is best supported.

**Comment by rdhar** [comment, marked as answer]
The easy answer is [one_dashboard_json](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/one_dashboard_json) since that's the "officially" recommended approach. In practice, this method melds well with importing and exporting raw dashboard JSON from the UI. That being said, you do need to be a bit more cognizant of handling the JSON in a separate file referenced with [templatefile](https://www.terraform.io/language/functions/templatefile) (which is documented with examples). 

This definitely supersedes [one_dashboard_raw](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/one_dashboard_raw) as that is a hybrid of both terraform and JSON mish-mashed together. It was introduced earlier as a stepping stone transitioning between the older REST API to GraphQL. 

Finally, [one_dashboard](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/resources/one_dashboard) was (I think) the first implementation of dashboards within the provider. If you don't want to mess around with JSON and have basic requirements that can be met with this limited approach, it should work just as well.
